com.blizzard.Hearthstone/com.blizzard.Hearthstone.yml

```diff
@@ -0,0 +1,107 @@
+app-id: com.blizzard.Hearthstone
+branch: stable
+
+runtime: org.winepak.Platform
+runtime-version: 3.0
+sdk: org.winepak.Sdk
+
+base: com.blizzard.BattleNet.BaseApp
+base-version: stable
+
+command: hearthstone
+
+inherit-extensions:
+  - org.winepak.Platform.Compat32
+  - org.winepak.Platform.Extension.corefonts
+  - org.winepak.Platform.Extension.vcrun2015
+
+add-extensions:
+  org.winepak.Platform.Wine:
+    directory: lib/wine
+    version: 3.9-staging
+    add-ld-path: lib
+    no-autodownload: false
+
+  org.winepak.Platform.Wine.Compat32:
+    directory: lib/wine-32bit
+    version: 3.9-staging
+    add-ld-path: lib
+    no-autodownload: false
+
+tags:
+  - proprietary
+
+finish-args:
+  - --socket=x11
+  - --socket=pulseaudio
+  - --share=ipc
+  - --share=network
+  - --device=dri
+  - --allow=multiarch
+  - --allow=devel
+
+modules:
+  - name: setup-wine
+    buildsystem: simple
+    build-commands:
+      - mkdir -p /app/lib/wine
+
+  - name: setup-wine-compat32
+    buildsystem: simple
+    build-commands:
+      - mkdir -p /app/lib/wine-32bit
+
+  - name: hearthstone
+    only-arches:
+      - x86_64
+    buildsystem: simple
+    build-commands:
+      - install -d /app/bin
+      - install hearthstone-installer /app/bin
+      - install hearthstone /app/bin
+      - install -Dm644 com.blizzard.Hearthstone.appdata.xml /app/share/appdata/com.blizzard.Hearthstone.appdata.xml
+
+    sources:
+      - type: script
+        dest-filename: hearthstone-installer
+        commands:
+          - if [ -z "$WINEPREFIX" ]; then
+          - '    echo "No wine prefix set or is empty, abort."'
+          - '    exit 1'
+          - fi
+          -
+          - source /app/bin/battlenet-installer "battlenet://Hearthstone"
+          -
+          - if [[ $? != 0 ]] ; then
+          - '    echo "Battle.net installer failed, abort."'
+          - '    exit 1'
+          - fi
+          -
+          - if [ -e "${WINEPREFIX}/dosdevices/c:/Program Files (x86)/Hearthstone" ] ; then
+          - '    echo "This prefix already has an exisiting ''Hearthstone'' install at ${WINEPREFIX}"'
+          - '    echo "In order to install ''Hearthstone'' you must move or delete the current prefix."'
+          - '    exit 1'
+          - fi
```

Comment:
Add
```
-
- echo "Installing Extension(s)..."
- source /app/lib/extension/d3dx9/bin/d3dx9-install64
- source /app/lib/extension/d3dx9/bin/d3dx9-install64-wow64
- 
- echo "Installer finished"
```

Reply:
Hum, it appears that the extension never creates those files. Setting `-force-d3d9` still seems to function on my machine with the current configuration - could you confirm?

Reply:
Hmm, I wonder if using Wine's native d3d9 libs are enough. Will test when I get a chance.